simulation/utils/generate_map.py

```diff
@@ -55,82 +57,101 @@ def get_seed(self):
+        return (C.y - A.y) * (B.x - A.x) < (B.y - A.y) * (C.x - A.x)
+
+    def are_lines_intersecting(self, i, j):
         """Function that checks if two lines ((X[i],Y[i]), (X[i+1], Y[i+1])), ((X[j],Y[j]), (X[j+1],Y[j+1])) intersect"""
         if i == j:
             return False
-        try:
-            return intersect(self.points_before_interpolation[i], self.points_before_interpolation[i + 1],
-                             self.points_before_interpolation[j], self.points_before_interpolation[j + 1])
-        except IndexError:
-            print(f'i = {i}, j = {j}, size = {len(self)}')
-
-    # ************
-    def exterminate_loops(self, step, y_res, ran=40):
+
+        A = self.points_before_interpolation[i]
+        B = self.points_before_interpolation[i + 1]
+        C = self.points_before_interpolation[j]
+        D = self.points_before_interpolation[j + 1]
+        order_1 = self.are_points_in_clockwise_order(A, C, D)
+        order_2 = self.are_points_in_clockwise_order(B, C, D)
+        order_3 = self.are_points_in_clockwise_order(A, B, C)
+        order_4 = self.are_points_in_clockwise_order(A, B, D)
+
+        return order_1 != order_2 and order_3 != order_4
+
+    
+    #STILL NOT CHANGED, WORKING ON IT
+    def delete_map_loops(self, step, angle, points_radius=50):
         """Function that gets rid of most of the loops in map so as to make it a little less crazy"""
         i = 0
         size = len(self.points_before_interpolation)
         while i < size - 1:
             i = size - 2 if i >= size - 1 else i
-            left_limit = i - ran if i >= ran else 0
-            right_limit = i + ran if i + ran < size else size - 2
+            left_limit = i - points_radius if i >= points_radius else 0
+            right_limit = i + points_radius if i + points_radius < size else size - 2
             for j in range(left_limit + 1, right_limit):
                 j = size - 2 if j >= size - 1 else j
-                if self.is_crossing(i, j):
+                if self.are_lines_intersecting(i, j):
                     del self.points_before_interpolation[i + 1]
-                    point = next_point(self.points_before_interpolation[size - 2], step, math.pi * 3 / 4, y_res)
-                    self.append(point)
+                    point = generate_next_point(self.points_before_interpolation[size - 2], step, angle, self.resolution[1])
+                    self.append_point_before_interpolation(point)
             i += 1
 
-    # ************
     def interpolate(self):
-        tck, u = splprep([self.X(), self.Y()], s=0)
+        tck, u = splprep([self.get_X_list(), self.get_Y_list()], s=0)
         x_array = np.linspace(0, 1, self.resolution[0])
         x_array, y_array = splev(x_array, tck, der=0)
-        x_array += self.x_offset
         self.map_points = np.vstack((x_array, y_array)).T
         self.segments = [(self.map_points[i-1], self.map_points[i]) for i in range(1, len(self))]
```

Comment:
```suggestion
        self.map_points = np.vstack((x_array, y_array)).T.to_list()
        self.segments = list(zip(self.map_points[:-1], self.map_points[1:]))
```
ewentualnie bez to_list, ale generalnie im mniej iterowania po elementach tym lepiej

Reply:
spoko, ja to później wszystko w ramach sugestii Maćka zmienię na numpy array

Reply:
później czyli pewnie dzisiaj, skoro jutro mamy się spotkać xd